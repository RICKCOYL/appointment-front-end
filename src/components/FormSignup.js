/* eslint-disable no-unused-vars */
/* eslint-disable eqeqeq */
/* eslint-disable no-console */
import { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { createUser } from '../api';

const FormSignup = () => {
  const [state, setState] = useState({
    username: '',
    email: '',
    password: '',
  });

  const dispatch = useDispatch();
  const auth = useSelector((state) => state.authReducer);

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(createUser(state));

    setState({
      username: '',
      email: '',
      password: '',
    });
  };

  if (auth.id) return <Redirect to="/booking" />;

  return (
    <div id="login-box">
      <form className="left" onSubmit={handleSubmit}>
        <input
          type="text"
          name="username"
          placeholder="Username"
          required
          onChange={(e) => setState({ ...state, username: e.target.value })}
          value={state.username}
        />
        <input
          type="text"
          name="email"
          placeholder="E-mail"
          required
          onChange={(e) => setState({ ...state, email: e.target.value })}
          value={state.email}
        />
        <input
          type="password"
          name="password"
          placeholder="Password"
          required
          onChange={(e) => setState({ ...state, password: e.target.value })}
          value={state.password}
        />

        <input type="submit" name="signup_submit" value="Sign up" />
        <br />
        <br />
        <a href="/">Already have an account</a>
      </form>

      <div className="right">

        <input type="submit" className="social-signin facebook" value="Sign up with facebook" />
        <input type="submit" className="social-signin twitter" value="Sign up with Twitter" />
        <input type="submit" id="google" className="social-signin google" value="Sign up with Google" />
      </div>
      <div className="or">OR</div>
    </div>
  );
};

export default FormSignup;
