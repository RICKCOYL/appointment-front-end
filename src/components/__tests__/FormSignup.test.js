import React from 'react';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import FormSignup from '../FormSignup';

configure({ adapter: new Adapter() });
const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <FormSignup />
  </Provider>,
);

describe('Signup Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });
  test('renders the Consultancies List Component', () => {
    const page = component.find('.Formbooking');
    expect(page).toMatchSnapshot();
  });
});
