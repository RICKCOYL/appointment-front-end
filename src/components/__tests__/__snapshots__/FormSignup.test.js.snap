// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component renders the Consultancies List Component 1`] = `ShallowWrapper {}`;
